### ISSUE TITLE
User membershipRequest showing unexpected behavior

### ISSUE THREAD

**Issue by Azad99-9** [PalisadoesFoundation/talawa-api#2242]
**Describe the bug**
before testing set org's `userRegistrationRequired: true`.

Three cases of unexpected behaviour related to membershipRequests

**case 1**. Everytime register with a fresh account, new `membershipRequest` is being added to org's membershipRequests field but not to user's membershipRequests field.

**case 2**. Even if the requests are being added to the org's membershipRequests field but they are not fetchable in the org's query.

**case 3**. As soon as we send a request through the sendMembershipRequest mutation all the previously added requests are also now fetchable.

**To Reproduce**

**Case 1:**
1. Register with a fresh account
2. check the org's membershipRequests field
3. check the new user's membershipRequests field
You will notice the membershipRequest is will be added to the org's membershipRequests field but the same will not be reflected in the user's membershipRequests field.

[case 1.webm](https://github.com/PalisadoesFoundation/talawa-api/assets/120930148/a783eea1-096c-4826-929b-c9984f00f825)

**Case 2:**
1. signup with multiple new accounts to a single org.
2. Now compare org's membershipRequests list in the database and the membershipRequests list fetched by the `organizations` query.
you will notice a mismatch of number of requests.

https://github.com/PalisadoesFoundation/talawa-api/assets/120930148/161d1161-e841-47f2-a722-fcaf96b6a737

Remember this org to reproduce case 3.

**Case 3:**
1. Signup with a fresh account by choosing an org different from the org used in case 2.
2. Now use sendMembershipRequest mutation and send a request to the org said in case 2.
3. Now compare org's membershipRequests list in the database and the membershipRequests list fetched by the `organizations` query.
you will notice the mismatch of number of requests disappears.

https://github.com/PalisadoesFoundation/talawa-api/assets/120930148/961e8ca9-d89e-4da9-8e1a-156200b8b1bd

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by AVtheking** [comment]
I would like to work on it .

**Comment by hg6658** [comment]
Assign me this issue please

**Comment by Azad99-9** [comment]
Hey @AVtheking any update on this?

**Comment by AVtheking** [comment]
Hi @Azad99-9  was busy due to my exams.. will start working on it from tomorrow.

**Comment by AVtheking** [comment]
Unassigning myself busy in some other issue.

**Comment by Chaitanya1672** [comment]
I would like to work on this issue

**Comment by Chaitanya1672** [comment]
Working on it

**Comment by Chaitanya1672** [comment]
I am unassigning myself from my due to being occupied with other commitments.

**Comment by syedali237** [comment]
Can I be assigned this issue?

**Comment by syedali237** [comment]
Working on it might take a few days..

**Comment by syedali237** [comment]
I've fixed case 1. Would you prefer I submit a PR for this now, or should I complete the fixes for the other two cases first and then make a single PR?

**Comment by Azad99-9** [comment]
please do everything in a single PR and share a video of it too.

**Comment by syedali237** [comment]
@Azad99-9 @palisadoes Should I raise a PR for case 1 of this issue?

**Comment by syedali237** [comment]
@varshith257 Should I go ahead and raise a PR for the fixes so far? I'm still working on the other two cases but wanted to check in.

**Comment by varshith257** [comment]
If you think needed multiple prs, go ahead

**Comment by palisadoes** [comment]
During the week of November 11, 2024 we will start a code freeze on the `develop` branches in Talawa, Talawa Admin and Talawa-API. 

We have completed a project to convert the Talawa-API backend to use PostgreSQL. Work will then begin with us merging code in the `develop` branches to a new `develop-postrgres` branch in each repository.

Planning activities for this will be managed in our #talawa-projects slack channel. A GitHub project will be created to track specially labeled issues. We completed a similar exercise last year using a similar methodology. 

Starting November 12, California time no new PRs will be accepted against the `develop` branch. They must be applied to the `develop-postrgres` branch. 

There are some GSoC project features that will need to be merged into develop. These will be the only exceptions.

This activity and the post GSoC 2024 start date was announced in our #general Slack channel last month as a pinned post.
